File: python-examples/17-order-processing/approve_order.py
"""Utility to approve an order pending fraud review.

Usage:
    python approve_order.py <execution_id>
    python approve_order.py <execution_id> --reject
"""

import asyncio
import os
import sys

from dotenv import load_dotenv
from polos import PolosClient

load_dotenv()


async def main():
    if len(sys.argv) < 2:
        print("Usage: python approve_order.py <execution_id> [--reject]")
        sys.exit(1)

    execution_id = sys.argv[1]
    approved = "--reject" not in sys.argv

    project_id = os.getenv("POLOS_PROJECT_ID")
    if not project_id:
        raise ValueError("POLOS_PROJECT_ID environment variable is required")

    client = PolosClient(
        project_id=project_id,
        api_url=os.getenv("POLOS_API_URL", "http://localhost:8080"),
    )

    print(f"{'Approving' if approved else 'Rejecting'} order...")
    print(f"  Execution ID: {execution_id}")

    await client.resume(
        suspend_execution_id=execution_id,
        suspend_step_key="fraud_review",
        data={"approved": approved},
    )

    print(f"Done! Order {'approved' if approved else 'rejected'}.")


if __name__ == "__main__":
    asyncio.run(main())

File: typescript-examples/17-order-processing/approve_order.ts
/**
 * Utility to approve or reject an order pending fraud review.
 *
 * Usage:
 *   npx tsx approve_order.ts <execution_id>
 *   npx tsx approve_order.ts <execution_id> --reject
 *
 * Environment variables:
 *   POLOS_PROJECT_ID - Your project ID (required)
 *   POLOS_API_URL - Orchestrator URL (default: http://localhost:8080)
 *   POLOS_API_KEY - API key for authentication (optional for local development)
 */

import 'dotenv/config';
import { PolosClient } from '@polos/sdk';

async function main(): Promise<void> {
  const args = process.argv.slice(2);

  if (args.length < 1) {
    console.log('Usage: npx tsx approve_order.ts <execution_id> [--reject]');
    process.exit(1);
  }

  const executionId = args[0]!;
  const approved = !args.includes('--reject');

  const projectId = process.env['POLOS_PROJECT_ID'];
  if (!projectId) {
    throw new Error(
      'POLOS_PROJECT_ID environment variable is required. ' +
        'Set it to your project ID (e.g., export POLOS_PROJECT_ID=my-project).',
    );
  }

  const client = new PolosClient({
    projectId,
    apiUrl: process.env['POLOS_API_URL'] ?? 'http://localhost:8080',
    apiKey: process.env['POLOS_API_KEY'] ?? '',
  });

  console.log(`${approved ? 'Approving' : 'Rejecting'} order...`);
  console.log(`  Execution ID: ${executionId}`);

  await client.resume(
    'order_processing_workflow',
    executionId,
    'fraud_review',
    { approved },
  );

  console.log(`Done! Order ${approved ? 'approved' : 'rejected'}.`);
}

main().catch(console.error);
